src/lib/dialog/dialog-ref.ts

```diff
@@ -14,10 +14,18 @@ export class MdDialogRef<T> {
   /** The instance of component opened into the dialog. */
   componentInstance: T;
 
+  /** Expose overlay backdrop click event */
+  backdropClicked: Observable<void>;
```

Comment:
Should be `backdropClick` (to match `OverlayRef`). 

I don't think there needs to be an API for `escapePressed`. Users should be able to add their own `keydown` or `keypress` handler to their dialog content.